### PR TITLE
FCN, TFCN, RFCN implementation and examples

### PR DESCRIPTION
**Pull request by lojzezust**
The FCN, TFCN and RFCN models have been refactored to work with the new pipeline. Example config files for running the training are also added.

Input methods for random input generation have been added. They generate random features and labels with the specified shapes.

`RandomClassificationInput` can be used to test classification models, where we want to predict a single label for each input example. It is used in the example for training the simple example model.

`RandomSegmentationInput` can be used to test segmentation models, where we want to predict an output mask. Input and output shapes can be specified. Should fit most segmentation models. It is used in examples for training FCN, RFCN and TFCN models.

### CODE REVIEW

File: configs/fcn_example.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            "bias_init": 0.0,
+            "padding": "VALID",
+            "pool_size": 2,
+            "pool_stride": 2
+        }
+    },
+    "task": {
+        "classname": "eoflow.tasks.TrainTask",
+        "config": {
+            "num_epochs": 10,
```
Comment (devisperessutti) [inline configs/fcn_example.json]: I would reduce to 2 to keep running times down. Same for the other examples

File: README.md [hunk 2]
```diff
@@ -1,189 +1,106 @@
...
-A toy example is available in `mains/fcn_main.py`. It uses the
-`configs/example.json` configuration file, the `data_loader/data_generator.py`,
-`models/example_model.py` and `trainers/example_trainer.py`.
+This repository provides a templated structure to generate TensorFlow projects. Using the same base structure for all TF projects should benefit model creation, debugging and experiments reproducibility.
+
+The project contains the package `eoflow` which contains the base abstract classes and implements the common models, tasks and input methods. Custom models tasks and input methods can also be implemented building on top of the provided abstract classes.
+
+The package can be installed by running the following command in the root directory of the project.
+```
+$ pip install eoflow
```
[the remark this answers is not in the document]
Comment (lojzezust) [reply README.md]: Yeah, you're right, `pip install .` is the correct way from the local repository.

I will add the second option as the default. Thanks, I didn't know it can be done this way.

File: eoflow/models/fcn_model.py [hunk 3]
```diff
@@ -1,35 +1,44 @@
-from .layers import conv1d, conv2d, deconv2d, crop_and_concat, max_pool_2d, weighted_cross_entropy
-from ..base import BaseModel
+import logging
 import tensorflow as tf
 import numpy as np
-import logging
+from marshmallow import Schema, fields
+
+from eoflow.base import BaseModel, ModelMode
```
[the remark this answers is not in the document]
Comment (lojzezust) [reply eoflow/models/fcn_model.py]: I think it has to be `from ..base import BaseModel, ModelMode` but I thought that looks confusing.
